code/WorkInProgress/Cael_Aislinn/Rust/fuel_compressor.dm

```diff
@@ -20,13 +62,103 @@ var/const/max_assembly_amount = 300
 	add_fingerprint(user)
 	/*if(stat & (BROKEN|NOPOWER))
 		return*/
-	interact(user)
+	if (!has_electronics)
+		interact(user)
 
 /obj/machinery/rust_fuel_compressor/attackby(obj/item/weapon/W as obj, mob/user as mob)
-	if (istype(W, /obj/item/weapon/rcd_ammo))
-		compressed_matter += 10
-		del(W)
-		return
+	if (istype(user, /mob/living/silicon) && get_dist(src,user)>1)
+		return src.attack_hand(user)
+	switch(has_electronics)// beginning the construction/deconstruction stuff
+		if(ASSEMBLED)
```

Comment:
Please you should take a look at construction datums, there's a page on on the repo's wiki.